rotkehlchen/history/events/structures/asset_movement.py

```diff
@@ -0,0 +1,200 @@
+import logging
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.accounting.mixins.event import AccountingEventType
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.location_details import get_formatted_location_name
+from rotkehlchen.exchanges.data_structures import hash_id
+from rotkehlchen.history.events.structures.base import (
+    HISTORY_EVENT_DB_TUPLE_WRITE,
+    HistoryBaseEntry,
+    HistoryBaseEntryType,
+)
+from rotkehlchen.history.events.structures.types import (
+    HistoryEventSubType,
+    HistoryEventType,
+)
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_fval
+from rotkehlchen.types import (
+    Location,
+    TimestampMS,
+)
+from rotkehlchen.utils.misc import ts_ms_to_sec
+
+if TYPE_CHECKING:
+    from more_itertools import peekable
+
+    from rotkehlchen.accounting.mixins.event import AccountingEventMixin
+    from rotkehlchen.accounting.pot import AccountingPot
+    from rotkehlchen.fval import FVal
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class AssetMovement(HistoryBaseEntry):
+    """Asset movement event representing deposits and withdrawals on exchanges."""
+
+    def __init__(
+            self,
+            sequence_index: int,
+            timestamp: TimestampMS,
+            location: Location,
+            event_type: HistoryEventType,
+            event_subtype: HistoryEventSubType,
+            asset: Asset,
+            balance: Balance,
+            location_label: str | None = None,
+            notes: str | None = None,
+            identifier: int | None = None,
+            event_identifier: str | None = None,
+            extra_data: dict[str, Any] | None = None,
```

Comment:
Same here. Would you need to provide extra data at construction time? Only thing I can think of is the potential memo some exchange deposit/withdrawals have for chains like monero. 

But then what's the point of giving some args as part of extra data and some as normal args in the constructor (address, transaction_id and exchange_uniqueid)? Do one or the other. This way the logic inside the constructor also becomes simpler.

Reply:
Added a `memo` argument and removed the `extra_data` argument. `memo` can be used as you mentioned and also for stuff like [this](https://github.com/rotki/rotki/blob/develop/rotkehlchen/data_import/importers/binance.py#L397) that was being put in `link` a few places.

Reply:
As discussed in discord, I've changed this to have the `extra_data` argument again, and removed most of the other args that were being put in `extra_data`. I've kept the `unique_id` arg, since it is needed for generating the event identifier, but its not being added to `extra_data` anymore. If the value being used for `unique_id` is needed in `extra_data` then it can be passed in there as well, likely under a different key such as `transaction_hash` or something.